mvc/models/model-binding.rst

```diff
@@ -1,8 +1,105 @@
+
+.. literalinclude:: model-binding/sample/src/MVCMovie/retro-sample.txt
+   :language: html
+
+Although the sample shows just two fields and doesn't even include validation, it takes only a moment to realize that the above code will quickly become repetive and cumbersome to write, even when it's just a few fields. Real world apps often have many fields combined with complex structures. Admittedly, this kind of mapping code is a bit boring to write as well. There are details on what's behind ASP.NET MVC's model binding abstraction in the next section, "How model binding works". 
+
+How model binding works
+-----------------------
+When ASP.NET MVC receives an incoming HTTP request, it routes it to a specific action method of a controller. It determines which action method to run based on what is in the route data, then it binds values from the HTTP request to that action method's parameters. For example, consider the following URL:
+
+`http://contoso.com/movies/edit/2`
+
+The default model binder compares the sections of the URL with the pattern defined in the ``RouteValues`` collection. Put a breakpoint in any action method to see the contents of the ``RouteData`` object at run time in a debug window, as illustrated here: 
+
+.. image:: model-binding/_static/routevalues-in-watch-window.png
+
+Since the  routing pattern looks like this, ``{controller=Home}/{action=Index}/{id?}`` (it's the default route), the route parts ``movies/edit/2`` map to the ``Movies`` controller, its ``Edit`` action method, which accepts an argument called ``id``. So the code for the action method should look something like this: 
+
+.. literalinclude:: model-binding/sample/src/MVCMovie/Controllers/MoviesController.cs
+   :language: c#
+   :lines: 63-76  
+   
+.. note:: The strings in the URL are not case sensitive.
+
+Now that routing knows which action method to use, it quickly checks for model binding attributes applied to the action method's arguments. If it finds a model binding attribute it will use the data source specified by the attribute. Otherwise, for each argument (a primitive type for now, more on complex types later), it looks in its data sources for something with a matching name. In the above example the only argument is an integer named `id`, so that is what the binder will look for. Below is a list of the data sources in the order that the model binder looks through them:
+ 
+#. ``Request.Form``: These are form values that go in the HTTP request using the POST method.
+#. ``RouteData.Values``: Contains information about each segment of the route. Route segments are separated by a forward slash ``/``.
+#. ``Request.QueryStrings``: Data to the right of a question mark (?) in a URL, separated by ampersands (&). For example, genre and subgenre are query strings: http://contoso.com/movies?genre=Drama&subgenre=Crime
```

Comment:
For example rather than passing ID as route data (http://contoso.com/movies/edit/2) you could use a QS http://contoso.com/movies/edit?id=2). 
[i'm not sure that ?id=2 is not route data - so you might have to rephrase it to differentiate /2 from the QS approach)